proto/proto/block_header.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+package block_header;
+
+import "digest.proto";
+
+message BlockHeader {
+    /// the hash of the previous blocks header.
+    digest.Digest prev_hash = 1;
+    /// a unique sequential number of the current block.
+    uint64 block_num = 2;
+    /// a commitment to an MMR of the entire chain where each block is a leaf.
+    digest.Digest chain_root = 3;
+    /// a commitment to account database.
+    digest.Digest account_root = 4;
+    /// a commitment to the nullifier database.
+    digest.Digest nullifier_root = 5;
+    /// a commitment to all notes created in the current block.
+    digest.Digest note_root = 6;
+    /// a commitment to a set of transaction batches executed as a part of this block.
+    digest.Digest batch_root = 7;
+    /// a hash of a STARK proof attesting to the correct state transition.
+    digest.Digest proof_hash = 8;
+    /// specifies the version of the protocol.
+    uint64 version = 9;
+    /// the time when the block was created.
+    uint64 timestamp = 10;
+    /// a sequential hash of all fields except the note_root.
+    digest.Digest sub_hash = 11;
+    /// hash of the sub_hash and the note_root.
+    digest.Digest hash = 12;
```

Comment:
These two field can probably be omitted as they can be computed from the above fields.

Reply:
good point, removed